it/tutorials/scala-for-java-programmers.md

```diff
@@ -0,0 +1,716 @@
+invece di quelli inclusi in `System.out`.
+
+#### Funzioni anonime
+
+Il codice precedente è semplice da capire e possiamo raffinarlo ancora
+un po’. Notiamo preliminarmente che la funzione `timeFlies` è definita
+solo per esser passata come argomento alla funzione `oncePerSecond`.
+Nominare esplicitamente una funzione con queste caratteristiche non è
+necessario. È più interessante costruire detta funzione nel momento in
+cui viene passata come argomento a `oncePerSecond`. Questo è possibile
+in Scala usando le *funzioni anonime*, funzioni cioè senza nome. La
+versione rivista del nostro programma timer usa una funzione anonima
+invece di *timeFlies* e appare come di seguito:
+
+    object TimerAnonymous {
+      def oncePerSecond(callback: () => Unit) {
+        while (true) { callback(); Thread sleep 1000 }
+      }
+      def main(args: Array[String]) {
+        oncePerSecond(() =>
+          println("time flies like an arrow..."))
+      }
+    }
+
+La presenza delle funzioni anonime in questo esempio è rivelata dal
+simbolo `=>` che separa la lista degli argomenti della funzione dal
+suo corpo. In questo esempio la lista degli argomenti è vuota e di fatti
+la coppia di parentesi sulla sinistra della freccia è vuota. Il corpo della
+funzione `timeFlies` è lo stesso del precedente.
+
+## Le Classi
+
+Come visto precedentemente Scala è un linguaggio orientato agli oggetti e
+come tale presenta il concetto di classe. (Per ragioni di completezza
+va notato che alcuni linguaggi orientati agli oggetti non hanno il concetto
+di classe; Scala non è uno di questi.) Le classi in Scala sono dichiarate
+usando una sintassi molto simile a quella usata in Java. Un'importante
+differenza è che le classi in Scala possono avere dei parametri. Questo
+concetto è mostrato nella seguente definizione dei numeri complessi.
+
+    class Complex(real: Double, imaginary: Double) {
+      def re() = real
+      def im() = imaginary
+    }
+
+Questa classe per i numeri complessi prende due argomenti, la parte
+immaginaria e quella reale del numero complesso. Questi possono esser
+passati quando si crea una istanza della classe `Complex` nel seguente
+modo: `new Complex(1.5, 2.3)`. La classe ha due metodi, `re` e `im` che
+danno l’accesso rispettivamente alla parte reale e a quella immaginaria
+del numero complesso.
+
+Da notare che il tipo di ritorno dei due metodi non è specificato esplicitamante.
+Sarà il compilatore che lo dedurrà automaticamente osservando la parte a destra
+del segno uguale dei metodi e deducendo che per entrambi si tratta di
+valori di tipo `Double`.
+
+Il compilatore non è sempre capace di dedurre i tipi come nel caso precedente;
+purtroppo non c’è una regola semplice capace di dirci quando sarà in grado di
+farlo e quando no. Nella pratica questo non è un problema poiché il compilatore
+sa quando non è in grado di stabilire il tipo che non è stato definito
+esplicitamente. Come semplice regola i programmatori Scala alle prime armi
```

Comment:
I'm not sure I agree with the suggestion of "getting used to omit types". While I understand the intent, I think this is leading readers to believe that in Scala we omit types in declarations as much as possible.